### PR TITLE
Modify generateDecorations implemenetation for wrap objects to attach to correct body/Frame

### PR DESCRIPTION
**Pull request by aymanhab**
In particular:

- Geometry created by WrapGeometry is now indicated as fixed.
- DecorativeGeometry is attached to the Frame to which it is connected using MobilizedBodyId from the Frame.

Fixes issue #1634

### Brief summary of changes
No API changes, Refactored implementation of generateDecorations of WrapObject subclasses

### Testing I've completed
GUI move sliders and wrapobjects move as expected
Updatied testVisualization to account for a WrapObject not in Ground. previous test had WrapObjects all in Ground to avoid numerical issues with comparing transforms.

### Looking for feedback on.
Documentation, test case.

### CHANGELOG.md (choose one)

- no need to update because no API has changed.

The Doxygen for this PR can be viewed at http://myosin.sourceforge.net/?C=N;O=D; click the folder whose name is this PR's number.

### CODE REVIEW

**Comment by aymanhab** [comment]
Test failure is unrelated time-out. Ready for review.

**Comment by aseth1** [comment]
The changes look reasonable to me for fixing the geometry to a `MobilizedBody`. However, seeing the same block of code repeated four times just screams to be implemented by the base `WrapObject`.

File: OpenSim/Simulation/Wrap/WrapCylinder.cpp [hunk 1]
```diff
@@ -820,13 +820,20 @@ void WrapCylinder::generateDecorations(bool fixed, const ModelDisplayHints& hint
         // assumptions between DecorativeCylinder aligned with y  and
         // WrapCylinder aligned with z
         ztoy.updR().setRotationFromAngleAboutX(SimTK_PI / 2);
-        const SimTK::Transform& X_GB = getFrame().getTransformInGround(state);
-        SimTK::Transform X_GW = X_GB*getTransform()*ztoy;
+        // B: base Frame (Body or Ground)
+        // F: PhysicalFrame that this ContactGeometry is connected to
```
Comment (chrisdembia) [inline OpenSim/Simulation/Wrap/WrapCylinder.cpp]: ContactGeometry -> WrapGeometry.

**Comment by aymanhab** [comment]
Thanks @aseth1 and @chrisdembia for the review. My understanding is that we're not changing the API so I only changed the implmentation. If that is not the case then I agree on refactoring the methods and moving common code to base class, can you please confirm the policy? Thanks.

**Comment by chrisdembia** [comment]
Oh, right. My opinion is that adding a protected function in this situation does not violate the policy.

**Comment by aseth1** [comment]
I agree with it being a protected utility for `WrapObejcts` with no change to the API. Thanks!

**Comment by aymanhab** [comment]
Thanks @chrisdembia :+1: for the quick turnaround. Test failure due to timeout.

**Comment by aseth1** [comment]
Confirmed failures are unrelated timeouts. Merging.

**Comment by aymanhab** [comment]
Thank you @aseth1

**Pull request merged**